### DISCUSSION TITLE
Async behavior and parallel flows

### DISCUSSION THREAD

**Discussion by pietz** [microsoft/promptflow#1551] [General]
All of my use cases are very light on resources because they basically just call external APIs to do the heavy lifting for them (LLMs, AI Search, ...). However, they oftentimes take several seconds to execute because they're waiting for the generation to complete.

So I think I can run pretty complex flows on very cheap hardware but it's clear the logic needs to support asynchronous behavior.

- How can I deploy my flows to get that?
- How can the execution of flows be made non-blocking?

Thanks!

**Comment by brynn-code** [comment]
Hello, I've created a feature ask issue for this :), please track here #1760
